### ISSUE TITLE
How do I access childviews of any layout? I want to change their values programatically. Thanks for building an awesome library.

### ISSUE THREAD

**Issue by javedk16** [j2esu/rv-joiner#12]

**Comment by kunal-jaggi** [comment]
I am also stuck here. I am trying to change the "text" attribute of child view programmatically but it is not working.

**Comment by j2esu** [comment]
Hello. If I understand right, the question is how to programmatically acess to JoinableLayout views. So, take a look at [Full Demo App](https://github.com/j2esu/rv-joiner/blob/master/full-demo-app/src/main/java/su/j2e/rvjoiner/demo/AMain.java) onCreateMethod. I use onInflateComplete callback to set on click listeners, like:

```
rvJoiner.add(new JoinableLayout(R.layout.clickable, new JoinableLayout.Callback() {
            @Override
            public void onInflateComplete(View view, ViewGroup parent) {
                view.findViewById(R.id.clickable_btn).setOnClickListener(AMain.this);
            }
        }));
```

"View view" param is the root of your JoinableLayout.

**Comment by j2esu** [comment]
NOTE: if you're using this method to save links to child views for later changing values (for example, text in TextView), you should be aware of RecyclerView behavior - it can recycle your layout and recreate it later, and all your changes will be lost, so ensure you also bind your current data in onInflateComplete callback

**Comment by kunal-jaggi** [comment]
This is a great library. The recycler view library works fine, expect one thing. Consider the following code snippet:

```
rvJoiner.add(new JoinableLayout(R.layout.movie_details));
rvJoiner.add(new JoinableLayout(R.layout.reviews));
```

In reviews.xml layout file I have a TextView with hard coded android:text="Reviews". But, in movie_details layout I want to dynamically populate some TextFields. For example, a movie title, release year, ratings etc. I am inflating the details xml and filling the elements from a DB cursor.

```
        View view = inflater.inflate(R.layout.movie_details, container, false);

        mMovieTileTxt = (TextView) view.findViewById(R.id.movieTitle);
        mMoviePoster =  (ImageView) view.findViewById(R.id.moviePoster);
```

And, then I pull values from a cursor and try to fill the views (using setText etc. APIs). But, this is not working. In the logs, I can see that cursor has the correct value but setText API does not change the text of a TextView.

**Comment by j2esu** [comment]
Of course, the view you've inflated are not displayed on the screen, because you've not added it. RvJoiner inflates your layout itself, and you should use inflate callback to find your text views, as i wrote in previous comment:

```
rvJoiner.add(new JoinableLayout(R.layout.movie_details, new JoinableLayout.Callback() {
            @Override
            public void onInflateComplete(View view, ViewGroup parent) {
                mMovieTileTxt = (TextView) view.findViewById(R.id.movieTitle);
                mMoviePoster =  (ImageView) view.findViewById(R.id.moviePoster);
            }
        }));
```

**Comment by kunal-jaggi** [comment]
It seems the onInflateComplete callback is never triggered. This throws a NullPointerException when I try to call setText on TextView.

```
        rvJoiner.add(new JoinableLayout(R.layout.movie_details, new JoinableLayout.Callback() {
            @Override
            public void onInflateComplete(View view, ViewGroup parent) {
                mMovieTileTxt = (TextView) view.findViewById(R.id.movieTitle);
                mMoviePoster =  (ImageView) view.findViewById(R.id.moviePoster);
                mMovieReleaseYear =  (TextView) view.findViewById(R.id.movieReleaseYear);
                mMovieRating =  (TextView) view.findViewById(R.id.movieRating);
                mMovieOverview =  (TextView) view.findViewById(R.id.movieOverview);
                mMovieFavorite =  (ImageView) view.findViewById(R.id.favoriteIcon);
            }
        }));
```

**Comment by kunal-jaggi** [comment]
Thanks. But I noticed something weird. The onInflateComplete callback is randomly triggered. The following code snippet shows how I am adding and getting references inside the onInflateComplete callback.

```

        rvJoiner.add(new JoinableLayout(R.layout.movie_details, new JoinableLayout.Callback() {
            @Override
            public void onInflateComplete(View view, ViewGroup parent) {
                mMovieTileTxt = (TextView) view.findViewById(R.id.movieTitle);
                mMoviePoster =  (ImageView) view.findViewById(R.id.moviePoster);
                mMovieReleaseYear =  (TextView) view.findViewById(R.id.movieReleaseYear);
                mMovieRating =  (TextView) view.findViewById(R.id.movieRating);
                mMovieOverview =  (TextView) view.findViewById(R.id.movieOverview);
                mMovieFavorite =  (ImageView) view.findViewById(R.id.favoriteIcon);
            }
        }));
```

And then later in my code I try to set the value of TextView using setText(string) API. I ran the app, it crashed and I got a NullPointerException. The OS killed the app. I restarted the app and again got a NPE. The third time, it worked. So, on first two launches the callback wasn't triggered and the objects were null.

**Comment by j2esu** [comment]
To be used in RecyclerView, my lib wraps your layout into RecyclerView.Adapter. This callback is called after RecyclerView.Adapter.onCreateViewHolder. It can be called at any time (in our case, at first time) when you scroll list and your layout became visible. So, you can you setText only after this callback. Moreover,

> NOTE: if you're using this method to save links to child views for later changing values (for example, text in TextView), you should be aware of RecyclerView behavior - it can recycle your layout and recreate it later, and all your changes will be lost, so ensure you also bind your current data in onInflateComplete callback

It seems you not clearly understand what happens inside of RecyclerView. [This video](https://www.youtube.com/watch?v=imsr8NrIAMs) can be useful. Good luck in your projects! :)

**Comment by kunal-jaggi** [comment]
Sorry, but one more thing. Inside a Fragment, I have overridden onCreateView (inside this callback I have onInflateComplete) and onLoadFinished. From within onLoadFinished, I call setText on TextView that I inflate within onInflateComplete. It seems that sometimes onLoadFinished is called before onInflateComplete (this is causing the NullPointerException). It's not consistent, sometimes, onLoadFinished is called after onInflateComplete. Is there a proven way to control the order of these callbacks?
